DapperDoodle/BaseSqlExecutor.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Data;
+using Dapper;
+using DapperDoodle.Interfaces;
+
+namespace DapperDoodle
+{
+    public class BaseSqlExecutor
+    {
+        public void InitialiseDependencies(IBaseSqlExecutorOptions options)
+        {
+            _connection = options.Connection;
+            Dbms = options.Dbms;
+        }
+
+        private IDbConnection _connection;
+        public DBMS Dbms { get; set; }
+
+        public T SelectQuery<T>(string sql, object parameters = null)
+        {
+            if (typeof(T) == typeof(int))
```

Comment:
so how does a client query for a collection of ints?, eg getting results for `select id from some_table order by created desc limit 10;` to get the last 10 created items?

Reply:
Is typeof(List<int>) also just int?